c++/triqs_cthyb/measures/density_matrix.cpp

```diff
@@ -56,7 +56,11 @@ namespace triqs_cthyb {
 
     z                          = mpi::all_reduce(z, c);
     block_dm                   = mpi::all_reduce(block_dm, c);
-    for (auto &b : block_dm) b = b / real(z);
+    // normalize by the weight and enforce the hermiticity of the density matrix
+    for (auto &b : block_dm){
+        auto b_sym = make_regular(0.5*(b + dagger(b)))/real(z);
+        b = b_sym;
+    }
```

Comment:
I would keep the normalization separate from the symmetrization
```suggestion
    for (auto &b : block_dm){
        // Normalize
        b /= real(z);
        
        // Enforce hermiticity
        b = make_regular(0.5*(b + dagger(b)));
    }
```